src/Microsoft.Data.SqlClient/netcore/src/Microsoft/Data/SqlClientX/PoolingDataSource.cs

```diff
@@ -307,18 +311,18 @@ private void CloseConnector(SqlConnector connector)
         /// </summary>
         internal readonly struct OpenInternalConnectionState
         {
-            internal readonly SqlConnectionX _owningConnection;
+            internal readonly SqlConnectionX? _owningConnection;
```

Comment:
My gut says these should be auto-properties... I think the general pattern is that member variables should always be private.

This might also change the requirements for having a constructor, such that you can use the object initializer syntax to create the instance instead.

It's six of one, half dozen of the other, but I think in general having member variables be private is good.